src/awkward/_connect/numba/arrayview.py

```diff
@@ -152,7 +153,17 @@ def to_numbatype(form):
 ########## Lookup
 
 
-@numba.extending.typeof_impl.register(ak._lookup.Lookup)
+class NumbaLookup(Lookup):
+    def __init__(self, layout, attrs, generator=None):
+        super().__init__(layout, generator=generator)
+        self._attrs = attrs
+
+    @property
+    def attrs(self):
+        return self._attrs
```

Comment:
@jpivarski I've opted to extend the `Lookup` class to attach `attrs` — I don't think this makes sense for the base class.

Are you OK with this approach?

Reply:
It didn't need to be a subclass—`Lookup` is already a "bag of stuff" and having more stuff (`attrs`) doesn't break the model.

A subclass is more complex than it needs to be, but it can be simplified later (not this PR).